src/axom/slam/Map.hpp

```diff
@@ -89,6 +89,42 @@ class Map : public MapBase<typename S::PositionType>, public StrPol
   using iterator = const_iterator;
   using iterator_pair = const_iterator_pair;
 
+private:
+  template <typename USet = SetType, bool HasValue = !std::is_abstract<USet>::value>
+  struct SetContainer;
+
+  template <typename USet>
+  struct SetContainer<USet, false>
+  {
+    SetContainer(const USet* set) : m_pSet(set) { }
+
+    const USet* get() const { return m_pSet; }
+
+    const USet* m_pSet;
+  };
+
+  template <typename USet>
+  struct SetContainer<USet, true>
+  {
+    SetContainer(const USet* set) : m_pSet(set) { }
+    SetContainer(const USet& set) : m_set(set) { }
+
+    const USet* get() const
+    {
+      if(m_pSet)
+      {
+        return m_pSet;
+      }
+      else
+      {
+        return &m_set;
+      }
+    }
+
+    const USet* m_pSet {nullptr};
+    USet m_set;
+  };
```

Comment:
This is nice! 
I wonder if it is more generally useful, e.g. outside of Map's implementation.